Fix bug where education major was not being detected in text searches

camerojo opened the pull request:
Also added in fix for small unrelated bug related to settings translation searches - wait icon was displaying as if search was underway even though search was not performed because it was invalid.

Commit:
Fix bug where education major was not being detected in text searches

Commit:
Fix minor bug where wait icon displays even though a search is invalid

sadatmalik commented on server/src/main/java/org/tbbtalent/server/model/es/CandidateEs.java:
```diff
@@ -289,7 +290,15 @@ public void copy(Candidate candidate, TextExtracter textExtracter) {
+                final String courseName = education.getCourseName();
+                if (courseName != null) {
+                    text = text == null ? courseName : text + " " + courseName;
+                }
```
I see why you're doing this. But it means the elastic search query will still fail in some cases -because the education major elastic search is specified as a term query, which will look for exact keyword matches.

We could change the way the search is setup in SavedSearchServiceImpl/computeElasticQuery(..) or fix the underlying candidate data before re-building the elastic search index. 

What do you think?

camerojo replied:
I tested this and it worked

sadatmalik replied:
Note we'll need to rebuild the elastic search index before testing.

But even so it will only work in specific cases - i.e. where the education major is null and the course name matches one of the education major drop-down values.

camerojo replied:
I'll take another look, but I did rebuild the ES and tested it and convinced myself that it worked. But maybe I missed something.

camerojo replied:
@sadatmalik I have tested and checked in a new version - can you take a look?

Commit:
Merge branch 'staging' into 192-normalise-education-major-drop-down-values

Commit:
No functional change - optimize imports

Commit:
Fix bug where nationality search type (and, or, not) was being added to educations

Commit:
Add new ElasticSearch field: educationMajors
Make existing educations field text - which just contains course names.

Change ElasticSearch education major filter to use new educationMajors field.
